### ISSUE TITLE
Create concept of a DeploymentTarget

### ISSUE THREAD

**Issue by seanhamlin** [uselagoon/lagoon#1946]
## Deprecate OpenShift wording

The idea here is to get Lagoon away from the notion that the cluster being deployed to is OpenShift, and update the API and database schema accordingly.

At the moment Lagoon can deploy to:

* OpenShift 3.11
* Native Kubernetes

But who is to say that in the future Fargate or ARO is not an option. Keeping it generic to `DeploymentTarget` gives us options.

## Allow splitting by `ENVIRONMENT_TYPE`

The idea here being to allow a given project to have its production environments deploy to 1 `DeploymentTarget` and non-production to another. The default would be to the same to keep backwards capability.

## Validate adding a new DeploymentTarget

When you add a new `DeploymentTarget` Lagoon should verify that this cluster is 'ready' for Lagoon to deploy into it. e.g. the cluster has:

* The Kubernetes API is reachable and valid
* The Lagoon remote Helm charts installed
* A test project can be deployed (e.g. Drupal example simple) and torn down

The `DeploymentTarget` might be in a `validating` phase whilst the above happens, and could stall any deployments.

## Work out the build and deploy system based on DeploymentTarget `Type`

At the moment you need to manually specify which Lagoon system needs to run for certain tasks

```
  -D, --activeSystemsDeploy string         Which internal Lagoon System is responsible for deploying
  -P, --activeSystemsPromote string        Which internal Lagoon System is responsible for promoting
  -R, --activeSystemsRemove string         Which internal Lagoon System is responsible for promoting
  -T, --activeSystemsTask string           Which internal Lagoon System is responsible for tasks
```

It would be nice if the DeploymentTarget could provide the correct one as a default.

e.g. Native k8s would be 

```
--activeSystemsDeploy lagoon_kubernetesBuildDeploy \
--activeSystemsPromote lagoon_openshiftBuildDeploy \
--activeSystemsRemove lagoon_kubernetesRemove \
--activeSystemsTask lagoon_kubernetesJob
```

**Comment by tobybellwood** [comment]
Adding to 2.0 milestone to cover deprecating hardcoded `Openshift` - other ACs are Post-2.0

**Comment by rocketeerbkw** [comment]
We also discussed allow to split deployments based on an environment name match/regex

**Comment by tobybellwood** [comment]
DeployTargets are now implemented in Lagoon in #2829
